starknet_py/net/full_node_client.py

```diff
@@ -659,6 +657,110 @@ async def get_contract_nonce(
         res = cast(str, res)
         return int(res, 16)
 
+    # ------------------------------- Trace API -------------------------------
+
+    async def trace_transaction(
+        self,
+        tx_hash: Hash,
+    ) -> TransactionTrace:
+        """
+        For a given executed transaction, returns the trace of its execution, including internal calls.
+
+        :param tx_hash: Hash of the executed transaction.
+        :return: Trace of the transaction.
+        """
+        res = await self._client.call(
+            method_name="traceTransaction",
+            params={
+                "transaction_hash": tx_hash,
+            },
+        )
+        return cast(
+            TransactionTrace, TransactionTraceSchema().load(res, unknown=EXCLUDE)
+        )
+
+    async def simulate_transactions(
+        self,
+        transactions: List[AccountTransaction],
+        skip_validate: bool = False,
+        skip_fee_charge: bool = False,
+        block_hash: Optional[Union[Hash, Tag]] = None,
+        block_number: Optional[Union[int, Tag]] = None,
+    ) -> List[SimulatedTransaction]:
+        # pylint: disable=too-many-arguments
+        """
+        Simulates a given sequence of transactions on the requested state, and generates the execution traces.
+        If one of the transactions is reverted, raises CONTRACT_ERROR.
+
+        :param transactions: Transactions to be traced.
+        :param skip_validate: Flag checking whether there are enough funds in the account.
+        :param skip_fee_charge: Flag deciding whether fee should be deducted from the balance before the simulation
+            of the next transaction.
+        :param block_hash: Block's hash or literals `"pending"` or `"latest"`
+        :param block_number: Block's number or literals `"pending"` or `"latest"`
+        :return: The execution trace and consumed resources for each transaction.
+        """
+        block_identifier = get_block_identifier(
+            block_hash=block_hash, block_number=block_number
+        )
+
+        simulation_flags = []
+        if skip_validate:
+            simulation_flags.append("SKIP_VALIDATE")
+        if skip_fee_charge:
+            simulation_flags.append("SKIP_FEE_CHARGE")
+
+        res = await self._client.call(
+            method_name="simulateTransactions",
+            params={
+                **block_identifier,
+                "simulation_flags": simulation_flags,
+                "transactions": [
+                    _create_broadcasted_txn(transaction=t) for t in transactions
+                ],
+            },
+        )
+        return cast(
+            List[SimulatedTransaction],
+            SimulatedTransactionSchema().load(res, unknown=EXCLUDE, many=True),
+        )
+
+    async def get_block_traces(
```

Comment:
Let's change it to `starknet_traceBlockTransactions` in snake case as in the spec